### ISSUE TITLE
Difflint's linters should be optional

### ISSUE THREAD

**Issue by BlackVegetable** [endlessm/difflint#8]
Difflint should not require any particular linters. While it should (and will) support a variety of linters and many languages, we should not be forcing users to have and use all of the linters we support for any particular language.  Instead, a configuration of which linters to use for which languages should be simple to edit.

I propose a configuration file titled `difflint.json` should be added describing which linters should be used for particular languages.  It would look like the following:

```js
{
    ".js": {
        "use eslint": true,
        "use jscs": true,
        "use jshint": false
    },
    ".py": {
        "use pep8": true,
        "use pyflakes": true
    }
}
```

(This example supposes the work I'm doing to get *eslint* to be supported is finished.)  In this example, all the current default linters would be enabled except for *jshint*.  This format gives the corner-case advantage of allowing for two different file formats for the same language to be associated different linters such as `.cpp` and `.c++`.  I'm not sure that is something anyone would ever want though, so perhaps the mapping should be from language to linters, not file extensions to linters.

Additionally, difflint should notify a user if a linter is enabled in the configuration but does not seem to be installed on the user's machine.  I propose this message should only be displayed if the linter notices a file with a relevant extension is being analyzed (I could be persuaded otherwise though.)

The `difflint -c` option could perform this analysis as well but would notify if any enabled linters are not present, even if the current repository is not using any files of those extensions.

A.C.

- [x] A configuration file is read to decide which linters to use for a given language or file extension
  - [x] Decide whether to map from extension or language to linters.
- [x] The current "required" linters are no longer required for the program to complete with a non-0 status code.
- [x] A error should be displayed if a supported linter is enabled but not installed at runtime.
  - [x] Difflint -c should have similar behavior.

**Comment by ptomato** [comment]
Here's my suggestion for a config file format, based on what I find convenient in Sublime Text:
```json
{
    "javascript": {
        "extensions": ["js"],
        "linters": ["jshint", "jscs"]
    },
    "python": {
        "extensions": ["py", "pyw"],
        "linters": ["pep8"]
    }
}
```

**Comment by Kurt-von-Laven** [comment]
+1

We also discussed that it might be nice either in this issue or a follow-up issue to have default values for extensions. This implies that there should be some documentation regarding the canonical names of recognized programming languages (e.g., do I type "JavaScript" or "javascript"? "python," "Python," or "Python3"?).

**Comment by Kurt-von-Laven** [comment]
Regarding configuration files, we discussed that for this issue it may be simplest to have a default configuration file in the data directory specified in setup.py and support for git repo-local configuration files. In a follow-up issue, a more complete configuration system could be added (if desired) that allows config files in parent directories all the way up to the user's home directory.

**Comment by BlackVegetable** [comment]
I'm relaxing this issue slightly to not include user-override values.  Obviously that is a feature we want but this PR will be large enough without that.  I'll make a followup issue for this once this is complete.

**Comment by Kurt-von-Laven** [comment]
Makes sense to me. Thanks for the update, and happy refactoring!

**Comment by BlackVegetable** [comment]
Also, given that, the README.md won't need to be updated for this issue.  I'll submit the PR now.

**Comment by BlackVegetable** [comment]
Up for review again @Kurt-von-Laven

**Comment by Kurt-von-Laven** [comment]
Ty @BlackVegetable; just 1 minor comment I missed the first time.

**Comment by BlackVegetable** [comment]
Whitespace removed from both files.  (On a new commit.)
